File: PHP/Factory/F0/Creator.php
<?php

abstract class Creator
{
    abstract public function factoryMethod(): Product;

    public function someOperation(): string
    {
        $product = $this->factoryMethod();
        $result = "Creator: The same creator's code has just worked with " .
            $product->operation();
        return $result;
    }
}

File: Csharp/Creational/Factory/Show/Show/Show/Creator.cs
namespace Show
{
    abstract class Creator
    {
        public abstract Product FactoryMethod();
        
        public string SomeOperation()
        {
            var product = FactoryMethod();
            var result = "Creator: The same creator's code has just worked with "
                         + product.Operation();

            return result;
        }
    }
}